pkg/clusteroperator/clusteroperator.go

```diff
@@ -0,0 +1,177 @@
+package clusteroperator
+
+import (
+	"context"
+	"time"
+
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	v1 "github.com/openshift/api/config/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+)
+
+const (
+	fiveScds time.Duration = 5 * time.Second
+	isTrue                 = "True"
+)
+
+// Builder provides struct for clusterOperator object.
+type Builder struct {
+	// ClusterOperator definition. Used to create a clusterOperator object.
+	Definition *v1.ClusterOperator
+	// Created clusterOperator object.
+	Object *v1.ClusterOperator
+	// apiClient opens api connection to the cluster.
+	apiClient *clients.Settings
+	// Used in functions that define or mutate clusterOperator definition. errorMsg is processed before the
+	// ClusterOperator object is created.
+	errorMsg string
+}
+
+// Exists checks whether the given clusterOperator exists.
+func (builder *Builder) Exists() bool {
+	if valid, _ := builder.validate(); !valid {
+		return false
+	}
+
+	glog.V(100).Infof("Checking if clusterOperator %s exists", builder.Definition.Name)
+
+	_, err := builder.apiClient.ClusterOperators().Get(
+		context.Background(),
+		builder.Definition.Name,
+		metaV1.GetOptions{})
+
+	return err == nil || !k8serrors.IsNotFound(err)
+}
+
+// IsAvailable check if the clusterOperator is Available.
+func (builder *Builder) IsAvailable() bool {
+	glog.V(100).Infof("Verify %s clusterOperator availability", builder.Definition.Name)
```

Comment:
nit:
either "Verify %s clusterOperator's availability"
or
"Verify the availability of %s clusterOperator"